node_pools.tf

```diff
@@ -99,6 +99,98 @@ resource "google_container_node_pool" "node_pool_mt" {
   }
 }
 
+resource "google_container_node_pool" "node_pool_dynamic_pods" {
+
+  provider = google-beta
+
+  # these can't be created or deleted at the same time.
+  depends_on = [google_container_node_pool.node_pool_platform]
+  # version    = data.google_container_cluster.primary.master_version
+  version = var.kube_version_gke
+
+  # We want the dynamic-pods node pool to be completely replaced
+  # instead of rolling deployment.
+  # The master_version will ensure that the node pool is created then
+  # destroyed if there is an update.
+  name = "${var.deployment_id}-dp-${formatdate("MM-DD-hh-mm", timestamp())}"
+
+  # this one can take a long time to delete or create
+  timeouts {
+    create = "30m"
+    update = "30m"
+    delete = "30m"
+  }
+
+  lifecycle {
+    create_before_destroy = true
+    ignore_changes        = [name]
+  }
+
+  location = var.zonal_cluster ? local.zone : local.region
+  cluster  = google_container_cluster.primary.name
+
+  # if we are 'regional' i.e. in 3 zones,
+  # "1" here means "1 in each zone"
+  initial_node_count = var.zonal_cluster ? "3" : "1"
+
+  autoscaling {
+    min_node_count = "0"
+    max_node_count = var.zonal_cluster ? var.max_node_count : ceil(var.max_node_count / 3)
+  }
+
+  management {
+    # https://cloud.google.com/kubernetes-engine/docs/how-to/node-auto-upgrades
+    # With this set to false, then an update will only occur when terraform runs
+    # because we set the node pool kubelet version to the version of the master,
+    # which will trigger an update, and the name including a timestamp will
+    # force a create then destroy event.
+    auto_upgrade = false
```

Comment:
set to true instead

Reply:
Wouldn't that cause the current running pods to be killed?

Reply:
Done

Reply:
if they don't have a grace period or a disruption budget, then yes. I think that it's acceptable for task to fail sometimes because of a node upgrade.